.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,37 @@
+name: Node.js SDK Continuous Build
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    name: Build Node ${{ matrix.node-version }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
+        node-version:
+          - 10.x
+          - 12.x
+          - 14.x
+          - 15.x
+    steps:
+      - name: Checkout AWS XRay SDK Node Repository @ default branch latest
```

Comment:
Can you add actions/cache? https://github.com/actions/cache/blob/main/examples.md#node---npm

Reply:
Or maybe this one, not sure why there are two sections for npm and lerna

https://github.com/actions/cache/blob/main/examples.md#node---lerna

Reply:
So I was hesitant to add caching because that was actually one fo the causes of failures in the flaky Travis tests: [this build](https://travis-ci.org/github/aws/aws-xray-sdk-node/builds/739341341/config) failed while [this build](https://travis-ci.org/github/aws/aws-xray-sdk-node/builds/739384919/config) passed and the only difference was disabling caching in the latter. The failed build was picking up an old version of a cached dependency.

However, I figure that has something to do with how lerna hoists dependencies so I'm sure the lerna-specific cache has it figured out. I'll add that.